service/src/main/java/uk/gov/hmcts/reform/fpl/service/removeorder/DraftOrderRemovalAction.java

```diff
@@ -40,17 +40,13 @@ public void remove(CaseData caseData, CaseDetailsMap data, UUID removedOrderId,
         Optional<Element<HearingOrdersBundle>> optionalHearingOrderBundle
             = caseData.getHearingOrderBundleThatContainsOrder(removedOrderId);
 
-        if (optionalHearingOrderBundle.isEmpty()) {
-            throw new IllegalStateException(format("Failed to find hearing order bundle that contains order %s",
-                removedOrderId));
-        }
-
-        Element<HearingOrdersBundle> selectedHearingOrderBundle = optionalHearingOrderBundle.get();
-        Element<HearingOrder> draftOrderElement = element(removedOrderId, draftOrder);
-
-        selectedHearingOrderBundle.getValue().getOrders().remove(draftOrderElement);
-        updateHearingOrderBundlesDrafts.update(
-            data, caseData.getHearingOrdersBundlesDrafts(), selectedHearingOrderBundle);
+        optionalHearingOrderBundle.map(selectedBundle -> {
```

Comment:
If you are doing it this way then it might be nicer to use the `ifPresentOrElse` method
An alternative is having it similar to how it was before but where you had
```java
        if (optionalHearingOrderBundle.isEmpty()) {
            throw new IllegalStateException(format("Failed to find hearing order bundle that contains order %s", removedOrderId));
        }
        Element<HearingOrdersBundle> selectedHearingOrderBundle = optionalHearingOrderBundle.get();
```
it  could be replaced with
```java
Element<HearingOrdersBundle> selectedHearingOrderBundle = optionalHearingOrderBundle.orElseThrow(() -> new IllegalStateException(format("Failed to find hearing order bundle that contains order %s", removedOrderId));

Reply:
sounds good. I updated just the above 2 lines with your suggestion and reverted the "update bundles" part as it was before.